trulens_eval/trulens_eval/tru.py

```diff
@@ -214,6 +239,51 @@ def __init__(
                 print(e)
                 self.db = OpaqueWrapper(obj=self.db, e=e)
 
+    @staticmethod
+    def _validate_and_compute_schema_name(name):
+        if not re.match(r"^[A-Za-z0-9_]+$", name):
+            raise ValueError(
+                "`name` must contain only alphanumeric and underscore characters!"
+            )
+        return f"TRULENS_APP__{name.upper()}"
+
+    @staticmethod
+    def _create_snowflake_database_url(
+        snowflake_connection_parameters: Dict[str, str], schema_name: str
+    ) -> str:
+        from snowflake.sqlalchemy import URL
+
+        Tru._create_snowflake_schema_if_not_exists(
+            snowflake_connection_parameters, schema_name
+        )
+        return URL(
+            account=snowflake_connection_parameters["account"],
+            user=snowflake_connection_parameters["user"],
+            password=snowflake_connection_parameters["password"],
+            database=snowflake_connection_parameters["database"],
+            schema=schema_name,
```

Comment:
can you remind me of why we are always computing schema on the fly and not allowing specifying them in conn params?

Reply:
because the schema will be decided based off the name of the "app" as per the eng doc.

Reply:
@sfc-gh-dkurokawa is it possible to add an option to use keypair auth instead of password? https://docs.snowflake.com/en/user-guide/key-pair-auth